### PR TITLE
Add ability to register a runner to the non-default self-hosted runne…

### PR DESCRIPTION
**Pull request by thechrisjohnson**
This adds the ability for a runner to be registered against a non-default self-hosted runner group if one exists.

### CODE REVIEW

File: src/Runner.Listener/Configuration/ConfigurationManager.cs [hunk 1]
```diff
@@ -159,17 +159,34 @@ public async Task ConfigureAsync(CommandSettings command)
...
+                agentPool = agentPools.Where(x => string.Equals(poolName, x.Name, StringComparison.CurrentCultureIgnoreCase) && !x.IsHosted).FirstOrDefault();
+            }
+            else
+            {
+                agentPool = defaultPool;
+            }
+
+            if (agentPool == null && poolName == null)
+            {
+                throw new TaskAgentPoolNotFoundException($"Could not find any self-hosted runner group. Contact support.");
```
Comment (joshmgross) [inline src/Runner.Listener/Configuration/ConfigurationManager.cs:181]:
```suggestion
                throw new TaskAgentPoolNotFoundException($"Could not find any self-hosted runner groups. Contact support.");
```

File: src/Runner.Listener/CommandSettings.cs [hunk 2]
```diff
@@ -169,6 +169,15 @@ public string GetRunnerName()
                 validator: Validators.NonEmptyValidator);
         }
 
+        public string GetRunnerGroupName(string defaultPoolName = null)
+        {
+            return GetArgOrPrompt(
+                name: Constants.Runner.CommandLine.Args.RunnerGroup,
+                description: "Enter the name of the runner group to add this runner to:",
```
Comment (joshmgross) [inline src/Runner.Listener/CommandSettings.cs:176]: Would it be worth listing the available runner groups that the user can choose?

Comment (thechrisjohnson) [reply src/Runner.Listener/CommandSettings.cs:176]: So we didn't in the past. I think the problem is around how many runner groups there are in the tenant.

File: src/Runner.Listener/Configuration/ConfigurationManager.cs [hunk 3]
```diff
@@ -159,17 +159,34 @@ public async Task ConfigureAsync(CommandSettings command)
...
-            TaskAgentPool agentPool = agentPools?.Where(x => x.IsHosted == false).FirstOrDefault();
+            TaskAgentPool defaultPool = agentPools?.Where(x => x.IsInternal).FirstOrDefault();
 
-            if (agentPool == null)
+            if (agentPools?.Where(x => !x.IsHosted).Count() > 1)
             {
-                throw new TaskAgentPoolNotFoundException($"Could not find any private pool. Contact support.");
+                poolName = command.GetRunnerGroupName(defaultPool?.Name);
+                _term.WriteLine();
+                agentPool = agentPools.Where(x => string.Equals(poolName, x.Name, StringComparison.CurrentCultureIgnoreCase) && !x.IsHosted).FirstOrDefault();
```
Comment (TingluoHuang) [inline src/Runner.Listener/Configuration/ConfigurationManager.cs:172]: Why current culture?

Comment (thechrisjohnson) [reply src/Runner.Listener/Configuration/ConfigurationManager.cs:172]: Whoops, good catch!

**Pull request merged**